content/blog/2024/06/2024-06-11-require-java-17.adoc

```diff
@@ -0,0 +1,125 @@
+In other words, you must bring your own Java package.
+One straightforward way to do this is to install Java 17 from your Linux distribution, as described on the package download site:
+
+https://pkg.jenkins.io/debian/[Debian]:: `apt-get install fontconfig openjdk-17-jre`
+https://pkg.jenkins.io/redhat/[Red Hat]:: `yum install fontconfig java-17-openjdk`
+https://pkg.jenkins.io/opensuse/[openSUSE]:: `zypper install dejavu-fonts fontconfig java-17-openjdk`
+
+By virtue of not requiring any custom repositories, this is certainly the simplest method (and the one used by the Jenkins project's https://github.com/jenkinsci/packaging/tree/7c74bdb1dc0e8f4790c4c8af3e3a4bf9bc6ee38a/molecule/default[packaging tests]),
+but it does not give the user a high degree of control over the Java runtime environment.
+As mentioned previously, the official Jenkins Docker images use Adoptium/Eclipse Temurin (as does the Jenkins infrastructure project).
+Enthusiastic users may wish to install Java from Adoptium or another vendor.
+Since 2021, Adoptium has provided Linux installation packages, as described in https://blog.adoptium.net/2021/12/eclipse-temurin-linux-installers-available/[a piece by George Adams].
+Ultimately, the choice of which Java vendor to use is your own, as long as that vendor provides Java 17 or Java 21.
+Refer to your chosen Java vendor for installation instructions.
+
+Once you have installed a suitable version of Java, configure Jenkins to use that Java runtime.
+The most straightforward way is to configure that version of Java as the default version of Java at the operating system (OS) level:
+
+https://pkg.jenkins.io/debian/[Debian]:: `update-alternatives --config java`
+https://pkg.jenkins.io/redhat/[Red Hat]:: `alternatives --config java`
+https://pkg.jenkins.io/opensuse/[openSUSE]:: `update-alternatives --config java`
+
+Alternatively, users who do not wish to change the default version of Java can customize the `JAVA_HOME` or `JENKINS_JAVA_CMD` environment variable as part of the Jenkins `systemd(1)` service unit.
+Refer to the link:/doc/book/system-administration/systemd-services/[Managing systemd services] section of the Jenkins documentation for more information.
+
+== Reporting issues
+
+If you find a regression in a plugin, please file a bug report in Jira:
+
+* https://issues.jenkins.io/browse/JENKINS-67907[JENKINS-67907: Java 17 Phase 3: Require Java 17 or newer]
+
+When reporting an issue, include the following information:
+
+. Use the https://issues.jenkins.io/browse/JENKINS-67907[JENKINS-67907] epic.
+. Provide the output of `java -version` (e.g., OpenJDK 64-Bit Server VM Temurin-21.0.3+9 build 21.0.3+9-LTS)
+. Provide the name, version, and architecture of the operating system you are using (e.g., Ubuntu 24.04.1 LTS x86_64).
+. Provide the _complete_ list of installed plugins as suggested in the link:/doc/book/system-administration/diagnosing-errors/#how-to-report-a-bug[bug reporting guidelines].
```

Comment:
When I clicked the "bug reporting guidelines" link, it followed a redirect but did not open to the location inside the page.

Avoid the redirect so that the page jumps directly to the location instead of redirecting and opening at the top of the page.
```suggestion
. Provide the _complete_ list of installed plugins as suggested in the link:/doc/book/troubleshooting/diagnosing-errors/#how-to-report-a-bug[bug reporting guidelines].
```